grants_tagger/evaluate_pubs.py

```diff
@@ -0,0 +1,127 @@
+import json
+import typer
+import torch
+
+from transformers import AutoModel, AutoTokenizer
+from tqdm import tqdm
+from sklearn.metrics import classification_report
+from sklearn.preprocessing import MultiLabelBinarizer
+from rich import print
+
+evaluate_pubs_app = typer.Typer()
+
+
+@torch.no_grad()
+def evaluate_pubs(
+    data_path: str, threshold: float = 0.5, batch_size: int = 4, device: str = "cpu"
+):
+    if not torch.cuda.is_available() and device == "cuda":
+        print("[bold red]CUDA is not available, using CPU")
+        device = "cpu"
+
+    with open(data_path, "r") as f:
+        data = json.load(f)
+
+    # Filter publications without tags
+    data = [pub for pub in data if pub["mesh_terms"]]
+
+    tokenizer = AutoTokenizer.from_pretrained("Wellcome/WellcomeBertMesh")
```

Comment:
this is hardcoded, ideally you should pass as parameter, you can name it `model_path` and make it to default to this, thats fine